test/NuGet.Core.FuncTests/Dotnet.Integration.Test/PackCommandTests.cs

```diff
@@ -81,6 +86,122 @@ public void CreatePackageWithFiles()
+                var workingDirectory = Path.Combine(testDirectory, projectName);
+                var projectFile = Path.Combine(workingDirectory, $"{projectName}.csproj");
+
+                _dotnetFixture.CreateDotnetNewProject(testDirectory.Path, projectName, " classlib", testOutputHelper: _testOutputHelper);
+
+                File.WriteAllText(Path.Combine(workingDirectory, "abc.txt"), "hello world");
+                File.WriteAllText(Path.Combine(workingDirectory, "abc.props"), "<project />");
+                File.WriteAllText(Path.Combine(workingDirectory, "abc.dll"), "");
+
+                using (var stream = new FileStream(projectFile, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+
+                    var anotherFrameworkItemGroup = @"<ItemGroup Condition="" '$(TargetFramework)' == 'net46' "">
+                    <Content Include=""abc.txt"" Pack=""True"" />
+                    </ItemGroup>";
+                    var netStandardItemGroup = @"<ItemGroup Condition="" '$(TargetFramework)' == 'netstandard1.4' "">
+                    <None Update=""abc.props"" Pack=""True""  PackagePath=""build"" />
+                    </ItemGroup>";
+                    var projectFrameworkItemGroup = @$"<ItemGroup Condition="" '$(TargetFramework)' == '{Constants.ProjectTargetFramework}' "">
+                    <None Include=""abc.dll"" Pack=""True""  PackagePath=""lib"" />
+                    </ItemGroup>";
+                    ProjectFileUtils.SetTargetFrameworkForProject(xml, "TargetFrameworks", $"netstandard1.4;{Constants.ProjectTargetFramework}");
+                    ProjectFileUtils.AddCustomXmlToProjectRoot(xml, anotherFrameworkItemGroup);
+                    ProjectFileUtils.AddCustomXmlToProjectRoot(xml, netStandardItemGroup);
+                    ProjectFileUtils.AddCustomXmlToProjectRoot(xml, projectFrameworkItemGroup);
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                _dotnetFixture.RestoreProjectExpectSuccess(workingDirectory, projectName, testOutputHelper: _testOutputHelper);
+                var results = _dotnetFixture.PackProjectExpectSuccess(workingDirectory, projectName, $"/p:PackageOutputPath={workingDirectory}", testOutputHelper: _testOutputHelper);
+
+                var nupkgPath = Path.Combine(workingDirectory, $"{projectName}.1.0.0.nupkg");
+                var nuspecPath = Path.Combine(workingDirectory, "obj", $"{projectName}.1.0.0.nuspec");
+
+                Assert.DoesNotContain("NU5118", results.AllOutput);
+                Assert.True(File.Exists(nupkgPath), "The output .nupkg is not in the expected place");
+                Assert.True(File.Exists(nuspecPath), "The intermediate nuspec file is not in the expected place");
+
+                using (var nupkgReader = new PackageArchiveReader(nupkgPath))
+                {
+                    var files = nupkgReader.GetFiles();
```

Comment:
Can we enforce all files?

For example, it seems like abc.txt should be included in contentFiles/net46/any/abc.txt, but it's not enforced here. 

https://github.com/NuGet/NuGet.Client/blob/1466fd7767c92b84fc4c22827ea9c3bf0ec82f5b/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/NuGetPackCommandTest.cs#L655-L673 does it better than this class. 

If all files were enumerated, it'll be easier to read and understand the test.